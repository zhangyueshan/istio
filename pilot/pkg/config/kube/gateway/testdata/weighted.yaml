apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  class: istio
  listeners:
  - hostname:
      match: Domain
      name: domain.example
    port: 80
    protocol: HTTP
    routes:
      resource: httproutes
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  namespace: default
spec:
  hosts:
  - hostnames: ["first.domain.example"]
    rules:
    - match:
        pathMatchType: Prefix
        path: /get
      action:
        forwardTo:
        - targetRef:
            name: httpbin
          weight: 2
        - targetRef:
            name: httpbin-other
          weight: 3
