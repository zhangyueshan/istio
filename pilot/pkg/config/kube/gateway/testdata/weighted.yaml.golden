apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: istio-system
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - '*.domain.example'
    port:
      name: http-80-gateway-gateway-istio-system
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - first.domain.example
  http:
  - match:
    - uri:
        prefix: /get
    route:
    - destination:
        host: httpbin.default.svc.cluster.local
      weight: 40
    - destination:
        host: httpbin-other.default.svc.cluster.local
      weight: 60
---
